crates/nargo/src/cli/verify_cmd.rs

```diff
@@ -43,16 +57,16 @@ fn process_abi_with_verifier_input(
             .clone();
 
         if !value.matches_abi(param_type) {
-            return Err(CliError::Generic(format!("The parameters in the main do not match the parameters in the {}.toml file. \n Please check `{}` parameter. ", VERIFIER_INPUT_FILE,param_name)));
+            return Err(CliError::Generic(format!("The value provided for {} does not match the type defined in the ABI.\n Please check the provided value.", param_name)));
         }
 
         match value {
             InputValue::Field(elem) => public_inputs.push(elem),
             InputValue::Vec(vec_elem) => public_inputs.extend(vec_elem),
             InputValue::Undefined => {
                 return Err(CliError::Generic(format!(
-                    "The parameter {} is not defined in the {}.toml file.",
-                    param_name, VERIFIER_INPUT_FILE
+                    "The parameter {} is not defined.",
```

Comment:
Why making the error message less precise?

Reply:
Because ABI encoding doesn't rely on the arguments coming from any particular place. This PR aims to separate any knowledge of the filesystem from the internal workings of proving/verifying so we can reuse this logic elsewhere where the passed arguments may not be loaded from a toml file.

Reply:
See https://github.com/noir-lang/noir/pull/454

Reply:
If we go with the AbiError route, you can have an AbiError saying the parameter is not defined, but then when converting to CliError, you would add the reference to the toml file.